### PR TITLE
CODEOWNERS: update based on o11y squad changes

### PR DESCRIPTION
**Pull request by gabor**
i did four things:
1. i added `/pkg/tsdb/prometheus @grafana/observability-metrics`
2. i added `/public/app/plugins/panel/logs @grafana/logs-traces-squad`
3. i removed `@grafana/observability-squad` from cloudwatch-related things
3. i searched for every instance of `@grafana/observability-squad` and replaced it with one of the three new squads

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -30,18 +30,18 @@ go.sum @grafana/backend-platform
 /scripts/drone/ @grafana/grafana-release-eng
 
 # Cloud Datasources backend code
-/pkg/tsdb/cloudwatch @grafana/cloud-datasources @grafana/observability-squad
```
Comment (toddtreece) [inline .github/CODEOWNERS:33]: did you intend to remove cloud datasources here as well?

**Comment by joey-grafana** [comment]
All good except for what Todd mentioned. 

Seems like we're undecided on the team names though so they might change!

File: .github/CODEOWNERS (hunk 1 above)
Comment (gabor) [reply .github/CODEOWNERS:33]: good catch, was not the plan 😄

**Review by connorlindsey — APPROVED** [review]

**Review by joey-grafana — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -98,14 +99,14 @@ go.sum @grafana/backend-platform
...
-/public/app/features/explore/ @grafana/observability-squad
+/public/app/features/explore/ @grafana/observability-experience-squad
 /public/app/features/plugins @grafana/plugins-platform-frontend
 /public/app/core/components/TransformersUI/spatial @grafana/grafana-edge-squad
 /public/app/plugins/panel/alertlist @grafana/alerting-squad
 /public/app/plugins/panel/barchart @grafana/grafana-bi-squad
 /public/app/plugins/panel/heatmap @grafana/grafana-bi-squad
 /public/app/plugins/panel/histogram @grafana/grafana-bi-squad
-/public/app/plugins/panel/nodeGraph @grafana/observability-squad
+/public/app/plugins/panel/nodeGraph @grafana/logs-traces-squad
```
Comment (ifrost) [inline .github/CODEOWNERS:109]: Could we add `/public/app/plugins/panel/logs` as well?

Comment (gabor) [reply .github/CODEOWNERS:110]: done!

**Review by ifrost — APPROVED** [review]

**Review by toddtreece — APPROVED** [review]

**Pull request merged**